src/components/QuranReader/TajweedBar/TajweedBar.tsx

```diff
@@ -0,0 +1,79 @@
+import { useEffect, useRef, useState } from 'react';
+
+import classNames from 'classnames';
+import useTranslation from 'next-translate/useTranslation';
+
+import styles from './TajweedBar.module.scss';
+
+import ChevronDownIcon from '@/icons/chevron-down.svg';
+
+const TAJWEED_COLORS = [
+  'edgham',
+  'mad-2',
+  'mad-2-4-6',
+  'mad-4-5',
+  'mad-6',
+  'ekhfa',
+  'qalqala',
+  'tafkhim',
+];
+
+const TajweedColors = () => {
+  const [showTajweedBar, setShowTajweedBar] = useState(true);
+  const [height, setHeight] = useState(0);
+  const ref = useRef(null);
+
+  const { t } = useTranslation('quran-reader');
+
+  const toggle = () => {
+    setShowTajweedBar(!showTajweedBar);
```

Comment:
React `useState` is asynchronous. To make sure we have the latest state when the use clicks multiple times back to back we should do:

```
setShowTajweedBar((prevShowTajweedBar) => !prevShowTajweedBar);
```